enterprise/models.py

```diff
@@ -3678,3 +3678,266 @@ def __str__(self):
+            "The date and time when the configuration was completed by the SSO orchestration."
+        )
+    )
+
+    validated_at = models.DateTimeField(
+        blank=True,
+        null=True,
+        help_text=_(
+            "The date and time when the configuration was validated and used for the first time."
+        )
+    )
+
+    # ---------------------------- SAP Success Factors attribute mappings ---------------------------- #
+
+    odata_api_timeout_interval = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_(
+            "SAP specific configuration: the timeout interval for the OData API in seconds."
+        ),
+        default=29
+    )
+
+    odata_api_root_url = models.CharField(
+        blank=True,
+        null=True,
+        max_length=255,
+        help_text=_(
+            "SAP specific configuration: the root url of the OData API."
+        )
+    )
+
+    odata_company_id = models.CharField(
+        blank=True,
+        null=True,
+        max_length=255,
+        help_text=_(
+            "SAP specific configuration: the company id of the OData API."
+        )
+    )
+
+    sapsf_oauth_root_url = models.CharField(
+        blank=True,
+        null=True,
+        max_length=255,
+        help_text=_(
+            "SAP specific configuration: the root url of the SAP SuccessFactors OAuth API."
+        )
+    )
+
+    odata_api_request_timeout = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_(
+            "SAP specific configuration: the timeout interval for the OData API in seconds."
+        ),
+        default=29
+    )
+
+    sapsf_private_key = models.TextField(
+        blank=True,
+        null=True,
+        help_text=_(
+            "SAP specific configuration: the private key used to sign the SAML assertion."
+        )
+    )
+
+    odata_client_id = models.CharField(
+        blank=True,
+        null=True,
+        max_length=255,
+        help_text=_(
+            "SAP specific configuration: the client id of the OData API."
+        )
+    )
+
+    oauth_user_id = models.CharField(
+        blank=True,
+        null=True,
+        max_length=255,
+        help_text=_(
+            "SAP specific configuration: the user id of the OAuth API."
+        )
+    )
+
+    # ---------------------------- history ---------------------------- #
+
+    history = HistoricalRecords()
+
+    def save(self, *args, **kwargs):
+        """
+        Override the save method to ensure that the configuration is locked once submitted and not completed by the
+        SSO orchestration api.
+        """
+        old_instance = EnterpriseCustomerSsoConfiguration.objects.filter(pk=self.pk).first()
+        # If we're not creating a record and the record has been submitted but not completed configuration
+        # ensure we lock configurable fields within the table.
+        if old_instance and self.submitted_at and not self.configured_at:
```

Comment:
we might wanna create a "pending_configuration?" kind of attribute and use that here - i imagine we may want to have a reusable, easy way to check if its pending